File: scripts/boot_check.ts
import { spawn } from "node:child_process";
import { join } from "node:path";
import { ROOT } from "../src/paths.js";

function run(argv: string[]): Promise<number> {
  return new Promise((resolve) => {
    const p = spawn(process.execPath, [join(ROOT, "node_modules/tsx/dist/cli.mjs"), join(ROOT, "src/cli.ts"), ...argv], {
      cwd: ROOT,
      stdio: "inherit",
    });
    p.on("exit", (c) => resolve(c ?? 1));
  });
}

const c0 = await run(["up"]);
console.error("UP_CODE", 0, c0);
const c1 = await run(["up"]);
console.error("UP_CODE", 1, c1);
const cs = await run(["status"]);
console.error("STATUS_CODE", cs);

File: scripts/boot_check.py
import json, sys
from ccop.__main__ import main

def run(argv):
    try:
        main(argv)
    except SystemExit as e:
        return e.code
    return 0

# up twice (idempotent)
for i in range(2):
    code = run(["up"])
    print("UP_CODE", i, code, file=sys.stderr)

code = run(["status"])
print("STATUS_CODE", code, file=sys.stderr)
